mlir/lib/Dialect/TTL/TTLOps.cpp

```diff
@@ -106,3 +107,21 @@ void TensorListInit::getCanonicalizationPatterns(RewritePatternSet &patterns,
                                                  MLIRContext *context) {
   patterns.add(std::make_unique<ListInitCanonicalizer>(context));
 }
+
+LogicalResult Return::verify() {
+  auto function = cast<func::FuncOp>((*this)->getParentOp());
```

Comment:
This would assert if the parent is not a `FuncOp`, right? 
Should we `dyn_cast` and verify instead, to emit a verifier error instead of an assertion?

Reply:
I had initially considered it, but wasn't sure if we wanted to limit ourselves to the `Func` dialect. But it probably makes sense, as other function-like operations seem unlikely for now.